### PR TITLE
Create FR and FP controllers inheriting from OdinController

### PR DESCRIPTION
**Pull request by jsouter**
Following discussion from https://github.com/odin-detector/odin-data/issues/346

Removes all logic relating to the frame processor flag from OdinDataController, and implements FR and FP specific logic in subclasses. If we do want to use the same controller for both the FP and FR adapter I have tested that it works okay with FastCS and it would be sufficient to delete the redundant (and now broken) frame_receiver_adapter.py

Also a small fix that renames a parameter in the Dummy decoders to class instead of name (as signals called "name" and "description" get excluded when the full ParameterTree is requested and can not be added to FastCS).

### CODE REVIEW

File: python/src/odin_data/control/odin_data_adapter.py [hunk 1]
```diff
@@ -85,7 +83,7 @@ def get(self, path, request):
         #    logging.error("{}".format(response))
         except ParameterTreeError as param_error:
             response = {
-                "response": "OdinDatatAdapter GET error: {}".format(param_error)
+                "response": "OdinDataAdapter GET error: {}".format(param_error)
```
Comment (jsouter) [inline python/src/odin_data/control/odin_data_adapter.py:86]: Realising this should probably get the class name instead if we're subclassing

Comment (GDYendell) [reply python/src/odin_data/control/odin_data_adapter.py:86]: Agree. Let's see what we decide first.

**Comment by GDYendell** [comment]
Alan's odin-control branch has been merged and deleted. I have update fastcs-dev to use odin-control main instead. If you rebase (and make sure to remove the commit to pin odin-control to a branch) it should fix the build here.

**Comment by jsouter** [comment]
> Alan's odin-control branch has been merged and deleted. I have update fastcs-dev to use odin-control main instead. If you rebase (and make sure to remove the commit to pin odin-control to a branch) it should fix the build here.

Okay pushed that, seems to work as before after rebasing.

**Review by ajgdls — CHANGES_REQUESTED** [review]
Would it be possible to break out the FrameProcessorController into its own file (so basically one file for each class, named the same as the class but with underscore rather than camel case)?  I think it will make it clearer to see what is available if you are just looking around the directories.

**Comment by jsouter** [comment]
> Would it be possible to break out the FrameProcessorController into its own file (so basically one file for each class, named the same as the class but with underscore rather than camel case)? I think it will make it clearer to see what is available if you are just looking around the directories.

Okay, just pushed that change

File: python/src/odin_data/control/odin_data_adapter.py (hunk 1 above)
Comment (GDYendell) [reply python/src/odin_data/control/odin_data_adapter.py]: OK so I think we change this to `self.__class__` now?

**Comment by GDYendell** [comment]
Note this will break fastcs-odin when merged, it will need to have its config files updated.

**Comment by GDYendell** [comment]
@jsouter could you squash the last 3 commits into one? I think we are good to merge.

**Comment by jsouter** [comment]
> @jsouter could you squash the last 3 commits into one? I think we are good to merge.

Done!

**Review by GDYendell — APPROVED** [review]

**Comment by GDYendell** [comment]
Uh, does this logic need to be moved somewhere else? https://github.com/odin-detector/odin-data/pull/362/files#diff-af879f9e3b965b6fe0832f413026c82c57dd5f10c0103d6c3a39723f68ff0cdfL41

**Comment by jsouter** [comment]
Ah good point, it's possible, I had assumed this was covered by OdinDataController.process_config_changes. I will try and test if the decoder gets reloaded when it should.

**Pull request merged**